staging/src/kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1/authorize.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2020 The CDI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"errors"
+
+	authentication "k8s.io/api/authentication/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog/v2"
+
+	"kubevirt.io/containerized-data-importer-api/pkg/apis/core"
+)
+
+const (
+	// AnnPrePopulated is a PVC annotation telling the datavolume controller that the PVC is already populated
+	AnnPrePopulated = core.GroupName + "/storage.prePopulated"
+	// AnnCheckStaticVolume checks if a statically allocated PV exists before creating the target PVC.
+	// If so, PVC is still created but population is skipped
+	AnnCheckStaticVolume = core.GroupName + "/storage.checkStaticVolume"
+)
+
+var ErrNoTokenOkay = errors.New("proceeding without token is okay under the circumstances")
+
+// Authorize indicates if the creating context is authorized to create the data volume
+// For sources other than clone (import/upload/etc), this is a no-op
+func (dv *DataVolume) Authorize(requestNamespace, requestName string, sarClient SubjectAccessReviewsProxy, nsClient NamespaceProxy, dsClient DataSourceProxy, userInfo authentication.UserInfo) (bool, string, cloneSourceHandler, error) {
```

Comment:
Maybe use the client native interfaces instead of our own proxies, so `SubjectAccessReviewsGetter`, `NamespacesGetter`, `DataSourcesGetter`?

Is that an improvement?

Reply:
Scratch that, can't have DataSourcesGetter in here (circular dep).  But maybe the others?  Or will that make the deps situation worse?

Reply:
Introduced a single `type AuthorizationHelperProxy interface`, hope that is fine

> I'm not that down on this
I am not sure if you meant you're not down with the approach or don't hate it too much